cloud/azure/deploy/topic.go

```diff
@@ -58,8 +59,21 @@ func (p *NitricAzurePulumiProvider) newEventGridTopicSubscription(ctx *pulumi.Co
 
 	subName := topicName + "-" + config.GetService()
 
-	_, err = eventgrid.NewEventSubscription(ctx, ResourceName(ctx, subName, EventSubscriptionRT), &eventgrid.EventSubscriptionArgs{
-		Scope: topic.ID(),
+	subscriptionId, err := random.NewRandomString(ctx, subName+"-ID", &random.RandomStringArgs{
+		Length:  pulumi.Int(EventSubscriptionRT.MaxLen),
+		Keepers: pulumi.Map{"subName": pulumi.String(subName)},
+		Upper:   pulumi.Bool(true),
+		Lower:   pulumi.Bool(true),
+		Special: pulumi.Bool(false),
+		Number:  pulumi.Bool(false),
+	})
+	if err != nil {
+		return err
+	}
+
+	_, err = eventgrid.NewEventSubscription(ctx, subName, &eventgrid.EventSubscriptionArgs{
```

Comment:
the URN will not be unique, replace subName with the unique name

Reply:
I think it should be, since the subName value doesn't need to be truncated in the urn

Reply:
this is a separate issue, logged here: https://github.com/nitrictech/nitric/issues/790

Reply:
This is already addressed by the provided subscriptionId (see EventSubscriptionName) is it not? (subscriptionId is a completed random string that uses the original subName for recalculation). The pulumi URN should be unique as there should not be more than one topic subscription per service

Reply:
We should throw an error if more than one subscription is added